sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/ReplicatedResourceClient.java

```diff
@@ -158,49 +151,67 @@ public Mono<StoreResponse> invokeAsync(RxDocumentServiceRequest request,
 
     private Mono<StoreResponse> getStoreResponseMonoWithSpeculation(RxDocumentServiceRequest request, Quadruple<Boolean, Boolean, Duration, Integer> forceRefreshAndTimeout) {
         CosmosEndToEndOperationLatencyPolicyConfig config = request.requestContext.getEndToEndOperationLatencyPolicyConfig();
+        AvailabilityStrategy strategy = config.getAvailabilityStrategy();
         List<Mono<StoreResponse>> monoList = new ArrayList<>();
-        List<RxDocumentServiceRequest> requests = new ArrayList<>();
 
-        if (speculativeProcessor.shouldIncludeOriginalRequestRegion()) {
-            monoList.add(getStoreResponseMono(request, forceRefreshAndTimeout));
-            requests.add(request);
+        if (strategy instanceof ThresholdBasedAvailabilityStrategy) {
```

Comment:
for write operation, we should only enable the speculative retry if the nonIdempotentWriteRetries is enabled